controllers/tasks/task.go

```diff
@@ -23,19 +23,34 @@ type TaskSpec struct {
 	KnownErrors []error
 }
 
+// TaskResponse represents the response of each task
+type TaskResponse struct {
+	Name     string
+	Response interface{}
+	Error    error
+}
+
 // Task is a specific task to be done by controller
 type Task interface {
-	Run() error
+	Run() (interface{}, error)
 }
 
 // RunAll executes all the Task in the given list of TaskSpec
-func RunAll(tasks []*TaskSpec) (string, error) {
+func RunAll(tasks []*TaskSpec) []*TaskResponse {
+	taskResp := []*TaskResponse{}
```

Comment:
Can we avoid pointer?

Reply:
Any specific reason? I used a pointer to avoid the high size of structs in return.

Reply:
I was worried about heap allocations.